_posts/2019-03-28-how-to-avoid-dirty-reads-with-event-store.md

```diff
@@ -0,0 +1,24 @@
+---
+layout: blog-post
+title: How to avoid Dirty Reads with Event Store
+author: Riccardo Di Nuzzo
+---
+Dirty Reads are not common but are possible with the default configuration. In this article we'll discuss why and consider how to avoid them.
+
+In brief, when a write is sent to an Event Store cluster, the master will write the events and replicate them to the other nodes in the cluster. The master will then wait for a quorum of nodes (cluster_size/2 + 1) to acknowledge the write before considering the write successful.
+
+In other words in a cluster of 3 nodes the quorum is 2 nodes. Before the Master can send back the acknowledge to the client it needs to wait the confirmation of write from at least 1 Slave node. For that reason **it is possible that a client can read from the other node that is not updated yet with the latest write**. This situation is defined as “Dirty Read”.
+
+If you must ensure that a client will always read the latest version of your data then you can...
+
+If you use the .net client api: use the connection setting\
+`PerformOnMasterOnly`
+
+If your client doesn’t provides this method use the connection string setting \
+`require-master=true`
```

Comment:
@riccardone I can't find this in the codebase or the docs, any more details?

Reply:
Most of the conn settings can be specified customising the connection string like:
"ConnectTo=tcp://admin:changeit@localhost:1113; HeartBeatTimeout=500;require-master=true"

https://eventstore.org/docs/dotnet-api/connecting-to-a-server/index.html#customising-connection-settings

Reply:
Ahh, the hyphen threw me, OK, cool 👍